### PR TITLE
feat: business entity registration

### PR DESCRIPTION
**Pull request by luciorubeens**
## Summary

- [x] Validate url from providers
- [x] Transform image/video format
- [x] Test all steps

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: src/tests/fixtures/coins/ark/transactions/business-registration.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		"version": 2,
+		"type": 2,
+		"typeGroup": 1,
+		"amount": "0",
+		"fee": "2500000000",
+		"sender": "DR2FTWeYxZZK7GqnUrwCeRVkMv6uXhrGzQ",
+		"senderPublicKey": "023b76bd1cc3ce9de335543e5b556e18242e902a529db173c03030b883b8f04111",
+		"recipient": "DR2FTWeYxZZK7GqnUrwCeRVkMv6uXhrGzQ",
+		"signature": "c572494c7fd4ce3fd010187f3573ee28faf80a9f1367e75ec0d444fbc7469a6b8829b5a71d5508f27e1a1cf48ba81739493a131c343ae78ed5315ad417be3ebb",
+		"asset": {
```
Comment (faustbrian) [inline src/tests/fixtures/coins/ark/transactions/business-registration.json:14]: https://github.com/ArkEcosystem/AIPs/blob/master/AIPS/aip-36.md#asset

**Comment by faustbrian** [comment]
The IPFS upload is now available in release 0.9.336 of the SDK.

https://github.com/ArkEcosystem/platform-sdk/commit/e2dc6c3048fcb697a819021acf286007f6c6718b#diff-36734e02609c632742c556f0e2b45067

File: src/domains/transaction/components/BusinessRegistrationForm/BusinessRegistrationForm.tsx [hunk 2]
```diff
@@ -321,7 +321,7 @@ export const BusinessRegistrationForm: RegistrationForm = {
 					// @TODO: let the user choose what sub-type they wish to use.
 					subType: Enums.EntitySubType.None,
 					// @TODO: use the name that the user entered. Has to be valid like a delegate username.
-					name: undefined,
+					name: ipfsData.meta.displayName,
```
Comment (faustbrian) [inline src/domains/transaction/components/BusinessRegistrationForm/BusinessRegistrationForm.tsx:324]: The name and display name are different values as stated by the comment. Something like `slugify(ipfsData.meta.displayName)` would work or letting the user enter the display name.

Comment (luciorubeens) [reply src/domains/transaction/components/BusinessRegistrationForm/BusinessRegistrationForm.tsx:324]: What about `constantCase` from `@arkecosystem/utils`?

Comment (faustbrian) [reply src/domains/transaction/components/BusinessRegistrationForm/BusinessRegistrationForm.tsx:324]: That would force the name to always be uppercase and not work as intended.

https://github.com/ArkEcosystem/core/blob/develop/packages/core-magistrate-crypto/src/transactions/utils/entity-schemas.ts#L19
https://github.com/ArkEcosystem/core/blob/develop/packages/core-magistrate-crypto/src/transactions/utils/entity-schemas.ts#L5-L10

**Review by faustbrian — CHANGES_REQUESTED** [review]
- The `meta` property in the `ipfsData` variable is always empty.
- The `ipfsData` variable contains keys with an `undefined` value which is invalid JSON. Keys with `undefined` values should be deleted before sending the file to IPFS.
- Multiple Source control and social media providers are missing.
- The name has no validation.
- The description has no validation.
- The website has no validation.

**Pull request merged**